kernel/mem/paging.cpp

```diff
@@ -14,12 +14,14 @@
 #include <mem/paging.hpp>
 #include <lib/bitmap.hpp>
 #include <lib/stdio.hpp>
+#include <lib/mutex.hpp>
 #include <dev/serial/rs232.hpp>
 #include <stddef.h>
 
 #define KADDR_TO_PHYS(addr) ((addr) - KERNEL_BASE)
 
 static uint32_t machine_page_count;
+static px_mutex_t mutex_paging;
```

Comment:
Can you double-check that these global constructors are getting called? I'm not sure we ever made sure that was happening.

Reply:
The global constructor is *not* getting called. I put a print statement in there and it didn't do anything, plus we never actually make a call to it in `boot.s` or in our C/C++ code anywhere.

Reply:
Is this something we really need to check? This is how you construct an object on the heap in C++

Reply:
#172 has been merged.